go/vt/vtgate/scatter_conn.go

```diff
@@ -363,15 +363,12 @@ func (stc *ScatterConn) StreamExecuteMulti(
 				qs    queryservice.QueryService
 			)
 			transactionID := info.transactionID
+			reservedID := info.reservedID
```

Comment:
not directly related to your changes, but we need to look over the mutex at the top of the method and check if we are using it correctly. Does not look right to me after skimming the code

Reply:
`ExecuteMultiShard` also doing the same thing, we can look at this together.

Reply:
nah, `ExecuteMultiShard` actually needs its mutex. I don't think `StreamExecuteMulti` needs it though